tasks/eva_2369/export_mongo_dbs.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2021 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import sys
+
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.mongodb import MongoDatabase
+
+logger = logging_config.get_logger(__name__)
+logging_config.add_stdout_handler()
+
+
+def dump_data_from_source(mongo_source: MongoDatabase, top_level_dump_dir):
+    try:
+        logger.info("Running mongodump from source...")
+
+        # Force table scan is performant for many workloads avoids cursor timeout issues
+        # See https://jira.mongodb.org/browse/TOOLS-845?focusedCommentId=988298&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-988298
+        mongo_source.dump_data(dump_dir=os.path.join(top_level_dump_dir, mongo_source.db_name),
+                               mongodump_args={"gzip": "", "forceTableScan": "", "numParallelCollections": "1"})
+    except Exception as ex:
+        logger.error(f"Error while dumping data from source!\n{ex.__str__()}")
+        sys.exit(1)
+
+
+def get_databases_list_for_export(file_path):
+    database_list = []
+    try:
+        with open(file_path) as file_object:
+            lines = file_object.readlines()
+    except FileNotFoundError:
+        print('Could not find file with database list to export. Please check file path and name.')
+    else:
+        for line in lines:
+            database_list.append(line.strip())
+
+    return database_list
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Dump data from a given MongoDB source',
+                                     formatter_class=argparse.RawTextHelpFormatter, add_help=False)
+    parser.add_argument("--mongo-source-uri",
```

Comment:
Mongodump/restore tools have the [nasty habit of displaying passwords in clear text](https://jira.mongodb.org/browse/TOOLS-1020) :(

So, unfortunately, we have to provide the URI without the password (may be this can be explicitly specified in the argument help text) and the secrets file (with just the password) separately. The [method that you suggest](https://github.com/EBIvariation/eva-tasks/blob/f32cec8bd498e02bc7ae4589e23b08c95633027e/tasks/eva_2287/correct_contig_with_chr_removed.py#L18) is good but it will put the password in the URI and then supply it to mongodump.

Reply:
Thanks for clarifying Sundar, let's stick to the current approach